hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -175,13 +175,19 @@ public BackgroundTaskQueue getTasks() {
   public List<ContainerBlockInfo> chooseContainerForBlockDeletion(
       int blockLimit, ContainerDeletionChoosingPolicy deletionPolicy)
       throws StorageContainerException {
+    long[] totalPendingBlockCount = new long[1];
```

Comment:
Why use an array here for a single variable?

Reply:
`totalPendingBlockCount` is an array of size 1 because variables used in lambda expressions must be effectively final. Using an array allows you to modify its content inside the lambda while keeping the array reference unchanged.